anyway/infographics_utils.py

```diff
@@ -78,7 +79,10 @@ class Widget:
     text: Dict
     meta: Optional[Dict]
 
-    def __init__(self, request_params: RequestParams, name: str):
+    def __init__(self, request_params: RequestParams, name: str = ""):
+        # name is mandatory only here. In the derived classes it is not needed The
+        # default is set to avoid type checking error.
+        assert name != ""
```

Comment:
I was trying to avoid the type checking error in line the second line
`    for w in get_widget_classes():
        widget: Widget = w(request_params)`
The type of w is `Widget`, which has two parameters for the constructor Here, name is mandatory. But actually, `w` is always one of the sub-classes, which has a single parameter to the constructor. 
Maybe is makes more sense to live with the type error...

Reply:
Think about using `Callable[[RequestParams], Widget]` instead of `Type[Widget]` in your functions - that will probably fix the type error and will better describe the code,
also try to delete the default value and the assert in `__init__`.

Reply:
Like this:
![image](https://user-images.githubusercontent.com/58947331/102672511-a2ad4c00-4199-11eb-9591-1dd0850f13c1.png)

Reply:
Thanks, good idea. I did not know about it.